core/metrics-core-service/src/main/java/org/hawkular/metrics/core/service/Order.java

```diff
@@ -55,6 +56,18 @@ public static Order fromText(String text) {
         return order;
     }
 
+    /**
+     * Determines a default value based on API parameters.
+     *
+     * @param limit maximum of rows returned
+     * @param start time range start
+     * @param end   time range end
+     * @return {@link #ASC} if limit is positive, start is not null and end is null; {@link #DESC} otherwise
+     */
+    public static Order defaultValue(int limit, Long start, Long end) {
```

Comment:
Because there is no simple default :)

Frankly I believe this is a bit confusing as well. But since the code for
that default was present in all the endpoints, I assumed it came from some
requirement. Maybe Stefan knows better ?
Le 29 avr. 2016 7:08 PM, "jsanda" <notifications@github.com> a écrit :

> In
> core/metrics-core-service/src/main/java/org/hawkular/metrics/core/service/Order.java
> <https://github.com/hawkular/hawkular-metrics/pull/497#discussion_r61611231>
> :
>
> > @@ -55,6 +56,18 @@ public static Order fromText(String text) {
> >          return order;
> >      }
> >
> > +    /**
> > +     * Determines a default value based on API parameters.
> > +     *
> > +     * @param limit maximum of rows returned
> > +     * @param start time range start
> > +     * @param end   time range end
> > +     * @return {@link #ASC} if limit is positive, start is not null and end is null; {@link #DESC} otherwise
> > +     */
> > +    public static Order defaultValue(int limit, Long start, Long end) {
>
> This is confusing. Why not simply default to either ascending or
> descending?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/hawkular/hawkular-metrics/pull/497/files/72769595c0bcf4863acb7ce85611bfc57eb13f3a#r61611231>
>